sources/packages/api/src/route-controllers/education-program-offering/education-program-offering.controller.ts

```diff
@@ -264,16 +268,21 @@ export class EducationProgramOfferingController extends BaseController {
     @Param("locationId") locationId: number,
     @Param("programId") programId: number,
     @Param("programYearId") programYearId: number,
+    @Query("selectedOfferingIntensity")
+    selectedOfferingIntensity: OfferingIntensity,
     @Query("includeInActivePY") includeInActivePY = false,
   ): Promise<OptionItem[]> {
     const offerings =
       await this.programOfferingService.getProgramOfferingsForLocation(
         locationId,
         programId,
         programYearId,
-        undefined,
+        selectedOfferingIntensity,
         includeInActivePY,
       );
+    if (!Object.values(OfferingIntensity).includes(selectedOfferingIntensity)) {
```

Comment:
This must happen before the call to the method `getProgramOfferingsForLocation`.